docs/IE_DG/Extensibility_DG/AddingNGraphOps.md

```diff
@@ -1,61 +1,63 @@
+4. Override the `clone_with_new_inputs` method, which enables graph manipulation routines to create copies of this operation and connect it to different nodes during optimization.
 
-5. Override the `visit_attributes` method, which allows serialization and deserialization of attributes. An `AttributeVisitor` is passed to the method, and the implementation is expected to walk over all the attributes in the op using the type-aware `on_attribute` helper. Helpers are already implemented for standard C++ types like `int64_t`, `float`, `bool`, `vector` and for existing nGraph defined types.
+5. Override the `visit_attributes` method, which enables serialization and deserialization of attributes. An `AttributeVisitor` is passed to the method, and the implementation is expected to walk over all the attributes in the op using the type-aware `on_attribute` helper. Helpers are already implemented for standard C++ types like `int64_t`, `float`, `bool`, `vector`, and for existing nGraph defined types.
 
 6. Override `evaluate`, which is an optional method that enables the application of constant folding if there is a custom operation on the constant branch.
 
-Based on that, declaration of a operation class can look as follows:
+Based on that, declaration of an operation class can look as follows:
 
 @snippet template_extension/op.hpp op:header
 
 ### Class Fields
 
 The provided implementation has several fields:
 
- * `add` of type `int64_t` is an attribute of custom operation
- * `type_info` of type `ngraph::NodeTypeInfo` defines the type and version of operation
+ * `add` of type `int64_t` is an attribute of a custom operation.
+ * `type_info` of type `ngraph::NodeTypeInfo` defines the type and version of an operation.
 
 ### Operation Constructors
 
-nGraph operation contains two constructors: a default constructor, which allows to create operation without attributes and a constructor that creates and validates operation with specified inputs and attributes.
+nGraph operation contains two constructors: 
+* Default constructor, which enables you to create an operation without attributes 
+* Constructor that creates and validates an operation with specified inputs and attributes
 
 @snippet template_extension/op.cpp op:ctor
 
 ### `validate_and_infer_types()`
 
-`ngraph::Node::validate_and_infer_types` method validates operation attributes and calculates output shapes using attributes of operation.
+`ngraph::Node::validate_and_infer_types` method validates operation attributes and calculates output shapes using attributes of the operation.
 
 @snippet template_extension/op.cpp op:validate
 
 ### `clone_with_new_inputs()`
 
-`ngraph::Node::clone_with_new_inputs` method creates a copy of nGraph operation with new inputs.
+`ngraph::Node::clone_with_new_inputs` method creates a copy of an nGraph operation with new inputs.
```

Comment:
Probably it is better to use `the` because we create a copy of the object for which the method is called.

Reply:
fixed